libs/auth-api-lib/src/lib/delegations/admin/delegation-admin-custom.service.ts

```diff
@@ -1,99 +1,226 @@
+  ): Promise<DelegationDTO> {
+    this.validatePersonsNationalIds(
+      delegation.toNationalId,
+      delegation.fromNationalId,
+    )
+
+    const zenDeskCase = await this.zendeskService.getTicket(
+      delegation.referenceId,
+    )
+
+    if (!zenDeskCase.tags.includes(DELEGATION_TAG)) {
+      throw new Error('Zendesk ticket is missing required tag')
+    }
+
+    if (zenDeskCase.status !== TicketStatus.Solved) {
+      throw new Error('Zendesk case is not solved')
+    }
+
+    const { fromReferenceId, toReferenceId } =
+      this.getNationalIdsFromZendeskTicket(zenDeskCase)
+
+    if (
+      fromReferenceId !== delegation.fromNationalId ||
+      toReferenceId !== delegation.toNationalId
+    ) {
+      throw new Error(
+        'Zendesk ticket nationalIds does not match delegation nationalIds',
+      )
+    }
+
+    const [fromDisplayName, toName] = await Promise.all([
+      this.namesService.getPersonName(delegation.fromNationalId),
+      this.namesService.getPersonName(delegation.toNationalId),
+    ])
+
+    return this.sequelize.transaction(async (transaction) => {
+      const newDelegation = await this.delegationModel.create(
+        {
+          id: uuid(),
+          toNationalId: delegation.toNationalId,
+          fromNationalId: delegation.fromNationalId,
+          createdByNationalId: user.actor?.nationalId ?? user.nationalId,
+          referenceId: delegation.referenceId,
+          toName,
+          fromDisplayName,
         },
-      ],
-    })
+        {
+          transaction,
+        },
+      )
 
-    const outgoingDelegations = await this.delegationModel.findAll({
-      where: {
-        fromNationalId: nationalId,
-      },
-      include: [
+      const ddt = await this.delegationDelegationTypeModel.create(
         {
-          model: DelegationScope,
-          required: true,
-          include: [
-            {
-              model: ApiScope,
-              required: true,
-              as: 'apiScope',
-              where: {
-                enabled: true,
-              },
-              include: [
-                {
-                  model: ApiScopeDelegationType,
-                  required: true,
-                  where: {
-                    delegationType: AuthDelegationType.Custom,
-                  },
-                },
-              ],
-            },
-          ],
+          delegationId: newDelegation.id,
+          delegationTypeId: AuthDelegationType.GeneralMandate,
+          validTo: delegation.validTo,
         },
-      ],
-    })
+        {
+          transaction,
+        },
+      )
 
-    return {
-      incoming: incomingDelegations.map((delegation) => delegation.toDTO()),
-      outgoing: outgoingDelegations.map((delegation) => delegation.toDTO()),
-    }
+      newDelegation.delegationDelegationTypes = [ddt]
+
+      // Index custom delegations for the toNationalId
+      void this.delegationIndexService.indexCustomDelegations(
```

Comment:
We need to update the index service to also index GeneralMandate delegations. It is currently only finding custom delegatoins.